sql/pre_install/tables.sql

```diff
@@ -333,16 +333,22 @@ SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.continuous_agg'
 -- See the comments for ContinuousAggsBucketFunction structure.
 CREATE TABLE _timescaledb_catalog.continuous_aggs_bucket_function (
   mat_hypertable_id integer NOT NULL,
-  -- The schema of the function. Equals TRUE for "timescaledb_experimental", FALSE otherwise.
-  experimental bool NOT NULL,
+  -- The schema of the function.
+  bucket_func_schema name NOT NULL,
   -- Name of the bucketing function, e.g. "time_bucket" or "time_bucket_ng"
-  name text NOT NULL,
+  bucket_func_name name NOT NULL,
   -- `bucket_width` argument of the function, e.g. "1 month"
   bucket_width text NOT NULL,
-  -- `origin` argument of the function provided by the user
-  origin text NOT NULL,
-  -- `timezone` argument of the function provided by the user
-  timezone text NOT NULL,
+  -- optional `origin` argument of the function provided by the user
+  bucket_origin text NULL,
+  -- optional `offset` argument of the function provided by the user
+  bucket_offset text NULL,
+  -- optional `timezone` argument of the function provided by the user
+  bucket_timezone text NULL,
+  -- fixed or variable sized bucket
+  bucket_fixed_width bool NOT NULL,
+  -- bucket defined on integer or time interval,
+  bucket_on_interval bool NOT NULL,
```

Comment:
If we save the REGPROCEDURE (aka Oid type) would be easy to infer it it is on INTEGER or INTERVAL bucket width (we have internal functions to handle it)

Reply:
Oid types have different lifetime guarantees as as the schema/name combination, this will require additional care in the update script as the oid will change on funcion signature changes